packages/atomic/src/components/atomic-breadbox/atomic-breadbox.tsx

```diff
@@ -0,0 +1,247 @@
+  @BindStateToController('facetManager')
+  @State()
+  public facetManagerState!: FacetManagerState;
+  @State() public error!: Error;
+  @State() private isCollapsed = true;
+
+  /**
+   * Number of breadcrumbs to display when collapsed.
+   */
+  @Prop() public collapseThreshold = 5;
+
+  public initialize() {
+    this.breadcrumbManager = buildBreadcrumbManager(this.bindings.engine);
+    this.facetManager = buildFacetManager(this.bindings.engine);
+  }
+
+  private limitPath(path: string[]) {
+    if (path.length <= 3) {
+      return path.join(SEPARATOR);
+    }
+
+    const ellipsedPath = [path[0], ELLIPSIS, ...path.slice(-2)];
+    return ellipsedPath.join(SEPARATOR);
+  }
+
+  private renderBreadcrumb(breadcrumb: Breadcrumb) {
+    const fullValue = Array.isArray(breadcrumb.formattedValue)
+      ? breadcrumb.formattedValue.join(SEPARATOR)
+      : breadcrumb.formattedValue;
+    const value = Array.isArray(breadcrumb.formattedValue)
+      ? this.limitPath(breadcrumb.formattedValue)
+      : breadcrumb.formattedValue;
+
+    return (
+      <li>
+        <Button
+          part="breadcrumb-button"
+          style="outline-neutral"
+          class="py-2 px-3 flex items-center btn-pill"
+          title={`${breadcrumb.label}: ${fullValue}`}
+          onClick={() => breadcrumb.deselect()}
+        >
+          <span
+            part="breadcrumb-label"
+            class="max-w-snippet truncate with-colon text-neutral-dark mr-px"
+          >
+            {breadcrumb.label}
+          </span>
+          <span part="breadcrumb-value" class="max-w-snippet truncate">
+            {value}
+          </span>
+          <atomic-icon class="w-2 ml-2" icon={CloseIcon}></atomic-icon>
+        </Button>
+      </li>
+    );
+  }
+
+  private renderShowMore(value: number) {
+    return (
+      <li>
+        <Button
+          part="show-more"
+          style="outline-primary"
+          text={`+ ${value.toLocaleString(this.bindings.i18n.language)}`}
+          class="p-2 btn-pill"
+          ariaLabel={this.bindings.i18n.t('show-n-more-filters', {value})}
+          onClick={() => (this.isCollapsed = false)}
+        ></Button>
+      </li>
+    );
+  }
+
+  private renderShowLess() {
+    return (
+      <li>
+        <Button
+          part="show-less"
+          style="outline-primary"
+          text={this.bindings.i18n.t('show-less')}
+          class="p-2 btn-pill"
+          onClick={() => (this.isCollapsed = true)}
+        ></Button>
+      </li>
+    );
+  }
+
+  private renderClearAll() {
+    return (
+      <li>
+        <Button
+          part="clear"
+          style="text-primary"
+          text={this.bindings.i18n.t('clear')}
+          class="p-2 btn-pill"
+          ariaLabel={this.bindings.i18n.t('clear-all-filters')}
+          onClick={() => this.breadcrumbManager.deselectAll()}
+        ></Button>
+      </li>
+    );
+  }
+
+  private get allBreadcrumbs(): Breadcrumb[] {
```

Comment:
There's a lot going on in this function :) 

I'm going to be boring, but just the usual "create N well named function that explains what processing were doing on each type of breadcrumb" would help make this more readable IMO.